### PR TITLE
Mypy

### PR DESCRIPTION
**Pull request by tjs-intel**
Enable Mypy and fix some issues.

### CODE REVIEW

File: sigoptlite/models.py [hunk 1]
```diff
@@ -71,10 +71,10 @@ class LocalTask:
 @dataclass(frozen=True, kw_only=True)
 class LocalParameterPrior:
   name: str
-  mean: float = None
-  scale: float = None
-  shape_a: float = None
-  shape_b: float = None
+  mean: float | None = None
```
Comment (harvey-intel) [inline sigoptlite/models.py:74]: I see, so the type for `mean` is float or None.

**Review by harvey-intel — APPROVED** [review]
Sure thing.

File: sigoptlite/models.py (hunk 1 above)
Comment (tjs-intel) [reply sigoptlite/models.py:74]: correct. If you think `None` should not be allowed I can remove `| None = None`

Comment (harvey-intel) [reply sigoptlite/models.py:74]: I don't really have a preference. cc @malkomes, what do you think here?

Comment (malkomes) [reply sigoptlite/models.py:74]: the `or` is fine by me.

**Pull request merged**